ksql-common/src/main/java/io/confluent/ksql/serde/FormatInfo.java

```diff
@@ -28,21 +28,23 @@
 
   private final Format format;
   private final Optional<String> avroFullSchemaName;
+  private final Optional<Character> delimiter;
 
   public static FormatInfo of(final Format format) {
-    return FormatInfo.of(format, Optional.empty());
+    return FormatInfo.of(format, Optional.empty(), Optional.empty());
```

Comment:
I think with the additional args being added to this that the set of factory methods would benefit from codifying the different valid permutations of parameters, e.g. something like:

```
public static FormatInfo json() {}
public static FormatInfo kafka() {}
public static FormatInfo avro(final Optional<String> fullSchemaName) {}
public static FormatInfo delimited(final Optional<Character> delimiter) {}
```

Reply:
This would only be used from tests, so I don't see the benefit of adding them.